tensorboard/webapp/metrics/store/metrics_store_internal_utils.ts

```diff
@@ -493,9 +504,40 @@ function getNextImageCardStepIndexFromSingleSelection(
   return null;
 }
 
+/**
+ * Gets next stepIndex for an image card based on range selection. Returns null if nothing should change.
+ * @param selectedSteps The selected steps from selected time. It should contain one or mutliple steps.
+ * @param steps The steps in an image card.
+ * @param step The step where the current step index locates.
+ */
+function getNextImageCardStepIndexFromRangeSelection(
+  selectedSteps: number[],
+  steps: number[],
+  step: number
+): number | null {
+  const firstSelectedStep = selectedSteps[0];
```

Comment:
If selectedSteps is [] then this will return undefined. What happens then?

Reply:
Good question, In practice all check will not pass and it returns null at the end. One test case is when selectedSteps is [].
Since you brought up I think it is not good to let it work as intented from side effect, I added a condition to return early.